### DISCUSSION TITLE
v23 alpha 2 无线网络链接逻辑错误

### DISCUSSION THREAD

**Discussion by xzl01** [linuxdeepin/developer-center#3872] [Troubleshooting | 故障排查] [answered]
环境：v23 aplha2 无线网卡
选择一个加密的wifi 进行链接，发现其不会自动弹出wifi密码输入界面，不符合操作逻辑

**Comment by xzl01** [comment]
cc @18202781743

**Comment by 18202781743** [comment]
v20上存在这个界面，可能是重构遗漏了这个逻辑，

**Comment by dfh1949** [comment]
我也遇到了，wpa_supplicant问题，试了几次，只有tp-link的wifi6才会这样，把wpa_supplicant换成iwd就直接没问题了

**Reply by xzl01** [reply]
这个可以提供下具体复现情况以便于我们排查问题，我更倾向于这并不是同一个问题

**Reply by dfh1949** [reply]
> 这个可以提供下具体复现情况以便于我们排查问题，我更倾向于这并不是同一个问题
系统：直接安装的V23_Alpha2
网卡:AX200 AX210
系统环境：初始化安装，未改动任何源和配置
路由器:WIFI6，TP-LINK或者可以直接手机热点，密码无，认证类型WPA-PSK/WPA2-PSK或WPA2-PSK/
相关包: wpasupplicant/2:2.10-8  network-manager/1.32.12.3-1+dde 
情况:
  1.直接链接wifi会要求输入密码，自动识别安全性为强制更改安全性为WPA-PSK/WPA2-PSK或WPA2-PSK 。
安全性改为无后才可以正常链接，之后仍然会隔一段时间要求输入密码。
 2.此情况只有在wifi6情况下才出现，wifi5没有问题
 3.此情况目前手头路由器只有TP-LINK的路由器才会有问题，其他品牌暂未发现。
其他品牌都会在密码改为无后路由器的认证类型会自动改为无，或者自动改为不加密，不会识别错误
 4.在直接进行iwd编译安装并替换wpasupplicant作为networkmanager后台上述问题解决
 5.将iwd作为networkmanager后台可以解决上述问题，但会导致玲珑包的浏览器无法联网，目前仅玲珑包的浏览器有问题，其余玲珑包软件无问题，没时间，未测试具体原因。

目前在几台电脑都可以复现，猜测是wpasupplicant问题，更换iwd都解决。此问题从V20后期测试到现在一直存在。

**Reply by ComixHe** [reply]
> 我也遇到了，wpa_supplicant问题，试了几次，只有tp-link的wifi6才会这样，把wpa_supplicant换成iwd就直接没问题了

bug已定位，是network-manager的关于wifi6的patch引起的错误

**Reply by ComixHe** [reply]
修复pr: https://github.com/deepin-community/network-manager/pull/2

**Comment by ComixHe** [comment]
更新信息，当tp-link的路由器重置后wifi应该是不设置密码和加密的，但目前v23无法正确识别对应的ap信息

**Comment by ComixHe** [comment, marked as answer]
> v23 alpha 2 无线网络链接逻辑错误

此bug在最新的dde-dock上现象已消失，可再次进行验证
